bpftune: init at unstable-2023-07-14, nixos/bpftune: init

NickCao opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

NickCao committed:
bpftune: init at unstable-2023-05-30

NickCao committed:
nixos/bpftune: init

NickCao committed:
nixos/bpftune: init basic test

SuperSandro2000 commented on pkgs/os-specific/linux/bpftune/default.nix:
```diff
@@ -0,0 +1,69 @@
+  '';
+
+  nativeBuildInputs = [
+    clang
```
Should we use clangStdenv?

Tungsten842 commented:
closes https://github.com/NixOS/nixpkgs/issues/240841

jian-lin commented:
@NickCao Is this signature broken?
![image](https://github.com/NixOS/nixpkgs/assets/75130626/f8074067-6d20-45d1-9c7b-43e4e380a991)

NickCao commented:
> @NickCao Is this signature broken?

Looks so, I'm on vacation and would handle it later.

NickCao commented on nixos/tests/bpftune.nix:
```diff
@@ -0,0 +1,37 @@
+    with subtest("bpftune tcp buffer size"):
+      server.wait_for_unit("iperf3.service")
+      client.succeed("iperf3 --reverse -c server")
+      client.wait_for_console_text("need to increase TCP buffer size")
```
We can always update the tests to account for later changes.

NickCao committed:
bpftune: unstable-2023-05-30 -> unstable-2023-07-11

Diff: https://github.com/oracle-samples/bpftune/compare/bfec6668db29ec83afe2b030440f4d8f1476c1a6...f2eb8e9c020261cc8c58b312841dec32d287a0cc

NickCao committed:
bpftune: unstable-2023-07-11 -> unstable-2023-07-14, enable parallel building

Diff: https://github.com/oracle-samples/bpftune/compare/f2eb8e9c020261cc8c58b312841dec32d287a0cc...66620152bf8c37ab592e9273fe87e567126801c2

SuperSandro2000 commented:
The test times out on aarch64

NickCao committed:
nixos/bpftune: drop flaky tests
